care/facility/api/viewsets/patient.py

```diff
@@ -599,19 +602,65 @@ def transfer(self, request, *args, **kwargs):
         return Response(data=response_serializer.data, status=status.HTTP_200_OK)
 
 
+class DischargePatientFilterSet(PatientFilterSet):
+
+    last_consultation_field = "last_discharge_consultation"
+
+    def filter_by_bed_type(self, queryset, name, value):
+        if not value:
+            return queryset
+
+        values = value.split(",")
+        filter_q = Q()
+        last_consultation_bed = (
+            ConsultationBed.objects.filter(end_date__isnull=False)
+            .order_by("consultation__patient_id", "-end_date")
+            .distinct("consultation__patient_id")
+            .values_list("id", "consultation_id")
+        )
+        if "None" in values:
+            filter_q |= ~Q(
+                last_discharge_consultation__id__in=[
+                    x[1] for x in last_consultation_bed
+                ]
+            )
+            values.remove("None")
+
+        if values:
+            filter_q |= Q(
+                last_discharge_consultation__id__in=ConsultationBed.objects.filter(
+                    id__in=[x[0] for x in last_consultation_bed],
+                    bed__bed_type__in=values,
+                ).values("consultation_id")
+            )
+
+        return queryset.filter(filter_q)
+
+
 @extend_schema_view(tags=["patient"])
 class FacilityDischargedPatientViewSet(GenericViewSet, mixins.ListModelMixin):
     permission_classes = (IsAuthenticated, DRYPermissions)
     lookup_field = "external_id"
     serializer_class = PatientListSerializer
     filter_backends = (
+        PatientDRYFilter,
         filters.DjangoFilterBackend,
         rest_framework_filters.OrderingFilter,
         PatientCustomOrderingFilter,
     )
-    filterset_class = PatientFilterSet
+    filterset_class = DischargePatientFilterSet
     queryset = (
-        PatientRegistration.objects.select_related(
+        PatientRegistration.objects.annotate(
+            last_discharge_consultation__id=Subquery(
```

Comment:
You dont need to explicitly specify it as a subquery, it should automatically understand that its a subquery.
ids = Something.objects.values_list('id', flat=True)
SomethingElse.objects.filter(something_id__in=ids)

Will only have one query.

Reply:
But will it not be required to use OuterRef?